src/runtime_config/default_actions.js

```diff
@@ -418,4 +421,49 @@ const actions = new Map([
       }
     ),
   ],
+  [
+    "transition_node_to_trusted",
+    new Action(
+      function (args) {
+        checkType(args.node_id, "string", "node_id");
+      },
+      function (args) {
+        const node = ccf.kv["public:ccf.gov.nodes.info"].get(
+          ccf.strToBuf(args.node_id)
+        );
+        if (node !== undefined) {
+          let node_obj = ccf.bufToJsonCompatible(node);
+          if (node_obj.status === "Pending") {
+            node_obj.status = "Trusted";
+            node_obj.ledger_secret_seqno = ccf.network.latestLedgerSecretSeqno();
+            ccf.kv["public:ccf.gov.nodes.info"].set(
+              ccf.strToBuf(args.node_id),
+              ccf.jsonCompatibleToBuf(node_obj)
+            );
+          }
+        }
+      }
+    ),
+  ],
+  [
+    "remove_node",
+    new Action(
+      function (args) {
+        checkType(args.node_id, "string", "node_id");
+      },
+      function (args) {
+        const node = ccf.kv["public:ccf.gov.nodes.info"].get(
+          ccf.strToBuf(args.node_id)
+        );
+        if (node !== undefined) {
```

Comment:
When is this the case? Are retired nodes deleted at some point? If not, then this check should go.

Reply:
Nodes aren't deleted now, but they will be. Regardless, a user may pass a node id for which there is no node?

Reply:
I don't think node deletion matters: The check prevents us writing something if we have a junk `node_id`, for a typo or a node which never existed. `remove_node(Bob)` should not write `Bob: status = Retired` to the KV.